client_test.go

```diff
@@ -175,13 +175,46 @@ func TestClientSyncBadID(t *testing.T) {
 	m := new(Msg)
 	m.SetQuestion("miek.nl.", TypeSOA)
 
+	c := &Client{
+		Timeout: 1 * time.Second,
+	}
+	if _, _, err := c.Exchange(m, addrstr); err == nil || !strings.Contains(err.Error(), "timeout") {
+		t.Errorf("query did not time out")
+	}
+	// And now with plain Exchange().
+	if _, err = Exchange(m, addrstr); err == nil || !strings.Contains(err.Error(), "timeout") {
```

Comment:
Checking the error message text here isn’t great, it would be better to use `errors.As` to get at the `net.Error` and check the `Timeout` method.

Also leave a TODO to instead use this 1.15 change when 1.14 support is dropped: https://golang.org/doc/go1.15#net.